### PR TITLE
Jrey/okta management

### PR DESCRIPTION
**Pull request by jreynard-code**
* [feat] Okta Client credentials auth flow management (environment variables: **OKTA_CLIENT_ID**, **OKTA_CLIENT_SECRET**, **OKTA_CLIENT_ISSUER**)
* [feat] Add log level management (environment variables: **LOG_LEVEL**, by default 'info')
* [feat] Add samples/run_with_okta.sh script to run fetch-scenario-parameters in docker container
* [feat] Add sample-with-okta and sample-ghcr-with-okta scripts in package.json
* [refactor] Add Utility class to file management and log management
* [refactor] Clients calls: as we manage now Aad and Okta Oauth clients, an IdentityProviderFactory is use to switch between client regarding environment variables (**IDENTITY_PROVIDER** = {aad/okta})
* [chore] Add deps (btoa:^1.2.1, axios:^0.25.0, request :^2.88.2,request-promise:^4.2.6, @cosmotech/api-ts:^0.1.1)
* [chore] Remove dep (@cosmotech/api)
* [chore] Adapt eslint configuration 
* [chore] Add .nvmrc file

### CODE REVIEW

File: src/constants/Constants.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  AZURE_IDENTITY_PROVIDER_NAME,
+  AZURE_TENANT_ID,
+  AZURE_CLIENT_ID,
+  AZURE_CLIENT_SECRET,
+  CSM_API_SCOPE,
+  OKTA_IDENTITY_PROVIDER_NAME,
+  OKTA_CLIENT_ID,
+  OKTA_CLIENT_SECRET,
+  OKTA_CLIENT_ISSUER,
+
```
Comment (csm-thu) [inline src/constants/Constants.js:44]: Style: newline should be removed

File: src/clients/OauthOktaClient.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * Oauth Client for Azure Active Directory
+ */
+class OauthOktaClient {
+  /**
+     * Default constructor
+     * @param {string} defaultScope the scope requested
+     */
+  constructor(defaultScope) {
+    this.defaultScope = defaultScope;
```
Comment (csm-thu) [inline src/clients/OauthOktaClient.js:17]: This auth client defines `defaultScope` for the scope property, whereas the Azure client defines `apiScope` for a similar value. Was that distinction done on purpose or can we use the same name to be more consistent ?

Comment (jreynard-code) [reply src/clients/OauthOktaClient.js]: Rename both defaultScope and apiScope in scope for consistency ;)

**Review by nborde-CSM — APPROVED** [review]
LGTM

**Review by mahkoCosmo — APPROVED** [review]

**Pull request merged**